src/main/java/de/tum/in/test/api/ast/model/JavaFile.java

```diff
@@ -0,0 +1,77 @@
+package de.tum.in.test.api.ast.model;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+
+/**
+ * Stores all required information about a Java file to be analysed
+ */
+public class JavaFile {
+	private static final PathMatcher javaFileMatcher = FileSystems.getDefault().getPathMatcher("glob:*.java");
+	private final Path javaFilePath;
+	private final CompilationUnit javaFileAST;
+
+	public JavaFile(Path javaFilePath, CompilationUnit javaFileAST) {
+		this.javaFilePath = javaFilePath;
+		this.javaFileAST = javaFileAST;
+	}
+
+	public Path getJavaFilePath() {
+		return javaFilePath;
+	}
+
+	public CompilationUnit getJavaFileAST() {
+		return javaFileAST;
+	}
+
+	/**
+	 * Turns the Java-file into an AST in case the provided path points to a
+	 * Java.-file
+	 *
+	 * @param pathOfFile Path to the Java-file
+	 * @return The information of the Java-file packed into a JavaFile object (null
+	 *         if the file is not a Java-file)
+	 */
+	public static JavaFile getJavaFileFromFileAt(Path pathOfFile) {
+		if (javaFileMatcher.matches(pathOfFile.getFileName())) {
+			try {
+				return new JavaFile(pathOfFile, StaticJavaParser.parse(pathOfFile));
+			} catch (IOException e) {
+				Logger.getLogger(Logger.GLOBAL_LOGGER_NAME).severe("IOException in JavaFile.getJavaFileFromFileAt("
```

Comment:
Define a logger-constant instead of calling `getLogger` everywhere.

Reply:
[Implemented part of the new feedback](https://github.com/ls1intum/Ares/pull/248/commits/ead5d75f0814dbd3738c57b058de8759b980e46d): Defined constant LOGGER (and updated constant names to all-capital).